src/main/scala/com/wavesplatform/history/HistoryWriterImpl.scala

```diff
@@ -111,17 +111,17 @@ class HistoryWriterImpl private(file: Option[File], val synchronizationToken: Re
       .reverse
   }
 
-  override def height(): Int = read { implicit lock => blockIdByHeight().size() }
+  override def height(): Int = lockfree { implicit lock => blockIdByHeight().size() }
 
   override def scoreOf(id: ByteStr): Option[BlockchainScore] = read { implicit lock =>
     heightOf(id).map(scoreByHeight().get(_))
   }
 
-  override def heightOf(blockSignature: ByteStr): Option[Int] = read { implicit lock =>
+  override def heightOf(blockSignature: ByteStr): Option[Int] = lockfree { implicit lock =>
```

Comment:
`NgHistoryReader`, which is actually used everywhere still acquires `read` here and in other places